cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -216,6 +222,8 @@ void SparsePoseGraph::ComputeConstraintsForScan(
   for (const mapping::Submap* submap : insertion_submaps) {
     const int submap_index = GetSubmapIndex(submap);
     CHECK(!submap_states_[submap_index].finished);
+    // however probability_grid might be final, so the last ray cast is drawn,
```

Comment:
I'm not sure I understand this comment. This code adds constraints between a submap and each of scans that were added to it.